File: macos/Classes/PlatformFileUrlPlugin.swift
import Cocoa
import FlutterMacOS

public class PlatformFileUrlPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "platform_file_url", binaryMessenger: registrar.messenger)
    let instance = PlatformFileUrlPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }

  func openFile(url:String){
         channel.invokeMethod("openFile", arguments:["url":url])
     }
}

File: android/src/main/kotlin/com/example/platform_file_url/PlatformFileUrlPlugin.kt
package com.example.platform_file_url

import androidx.annotation.NonNull

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
//可以优化 参考 https://github.com/llfbandit/app_links/blob/master/app_links/android/src/main/java/com/llfbandit/app_links/AppLinksPlugin.java
/** PlatformFileUrlPlugin */
class PlatformFileUrlPlugin() : FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel

  init {
    platformFileUrlPlugin=this
  }


  companion object {
    var platformFileUrlPlugin : PlatformFileUrlPlugin? = null
  }

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "platform_file_url")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
    if (call.method == "getPlatformVersion") {
      result.success("Android ${android.os.Build.VERSION.RELEASE}")
    } else {
      result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }

  fun openFile(url: String) {
    channel?.invokeMethod("openFile", mapOf("url" to url))
  }



}
